sdk/core/azure-core/src/http/raw_response.cpp

```diff
@@ -39,20 +42,19 @@ void RawResponse::AddHeader(uint8_t const* const begin, uint8_t const* const las
   end = std::find(start, last, '\r');
   auto headerValue = std::string(start, end); // remove \r
 
-  AddHeader(headerName, headerValue);
+  return AddHeader(headerName, headerValue);
 }
 
-void RawResponse::AddHeader(std::string const& header)
+size_t RawResponse::AddHeader(std::string const& header)
 {
   return AddHeader(
       reinterpret_cast<uint8_t const*>(header.data()),
       reinterpret_cast<uint8_t const*>(header.data() + header.size()));
 }
 
-void RawResponse::AddHeader(std::string const& name, std::string const& value)
+size_t RawResponse::AddHeader(std::string const& name, std::string const& value)
 {
-
-  this->m_headers.insert(std::pair<std::string, std::string>(name, value));
+  return Details::InsertHeaderWithValidation(this->m_headers, name, value);
```

Comment:
throw invalid argument if this fails?

Reply:
I would expect caller to decide instead, if they want to throw.
This API is expected to be used by Creating a Request or Response. This means it will be either a Service (storage) or a Transport adapter.
Now, if we throw here a runtime_error, the caller would not be able to just bubble up the exception because, for example, the transport adapter would need to document and depend on this type of exception.
That's why I think it is better to let the caller decide if they want to throw when the return value here is greater than 0

Reply:
If this is a general purpose utility method, only meant for internal use, and you are punting the decision to throw to the caller (in which case our client APIs or other public HTTP APIs), then that's fine.